nltk/translate/chrf_score.py

```diff
@@ -75,18 +84,31 @@ def sentence_chrf(reference, hypothesis, min_len=1, max_len=6, beta=3.0):
     :type max_len: int
     :param beta: the parameter to assign more importance to recall over precision
     :type beta: float
+    :param ignore_whitespace: ignore whitespace characters in scoring
+    :type ignore_whitespace: bool
     :return: the sentence level CHRF score.
     :rtype: float
     """
     return corpus_chrf([reference], [hypothesis], min_len, max_len, beta=beta)
 
 
-def corpus_chrf(list_of_references, hypotheses, min_len=1, max_len=6, beta=3.0):
+def _preprocess(sent, ignore_whitespace):
+    if type(sent) != str:
+        # turn list of tokens into a string
+        sent = ' '.join(sent)
+
+    if ignore_whitespace:
+        sent = re.sub(r'\s+', '', sent)
+    return sent
+
+
+def corpus_chrf(references, hypotheses, min_len=1, max_len=6, beta=3.0,
+                ignore_whitespace=True):
     """
     Calculates the corpus level CHRF (Character n-gram F-score), it is the
-    micro-averaged value of the sentence/segment level CHRF score.
+    macro-averaged value of the sentence/segment level CHRF score.
 
-    CHRF only supports a single reference.
+    This implementation of CHRF only supports a single reference at the moment.
```

Comment:
I think it's possible to support multiple references. From the paper, it's unclear but from the code, it looks like the default is to use the best score https://github.com/m-popovic/chrF/blob/master/chrF%2B%2B.py#L155

Reply:
Yes, I suggested it in the related issue as well: https://github.com/nltk/nltk/issues/1785

Would it make sense to implement this behavior as part of the current PR? That means that `chrf3` would have the same interface as e.g. BLEU, is that correct?

I didn't want to break the current interface so I left this unimplemented but I'm happy to do it this way.